test/python/unittest/API/APITest.py

```diff
@@ -0,0 +1,322 @@
+"""
+This file is part of the openPMD-api.
+
+Copyright 2018 openPMD contributors
+Authors: Axel Huebl, Carsten Fortmann-Grote
+License: LGPLv3+
+"""
+
+import openPMD
+
+import os
+import shutil
+import unittest
+try:
+    import numpy as np
+    found_numpy = True
+except ImportError:
+    found_numpy = False
```

Comment:
Ah yes, a simple runtime ```try```-```except``` to turn off features instead of CMake magic with ```#ifdef``` conditional compilation. :slightly_smiling_face:

Reply:
What could possibly go wrong... ;)